routes/volunteers_routes.js

```diff
@@ -0,0 +1,209 @@
+        })
+        .then((vol_info_col) => {
+            res.status(200).json(vol_info_col);
+        })
+        .catch(err => res.status(403).json({ message: err }));
+
+};
+///POST volunteer/department/department_id/volunteers
+var post_volunteers = function (req, res) {
+    return __has_permissions(1, VOLUNTEER_DEPT_MANAGER)
+        .then(() => {
+            var mail_addresses = req.body.map(vol_add => vol_add.email);
+            var result = [];
+            DrupalAccess.get_user_by_email(mail_addresses).then(users_data => {
+                Promise.all((mail_addresses.map((mail_addr) => {
+                    var data_to_save = users_data ? users_data.find((udata) => udata.email === mail_addr) : undefined;
+                    if (data_to_save === undefined || data_to_save.user_data === undefined) {
+                        result.push({ email: mail_addr, status: 'NotFound' })
+                    } else {
+                        vol_data = req.body.find(x => x.email === mail_addr);
+                        return Volunteer.forge({
+                            user_id: data_to_save.user_data.uid,
+                            department_id: req.params.department_id,
+                            role_id: req.params.role_id,
+                            event_id: CURRENT_EVENT
+                        }).save().then((save_result) => {
+                            log.info('Saved ' + JSON.stringify(save_result));
+                            result.push({ email: mail_addr, status: 'OK' });
+                        }).catch((err) => {
+                            result.push({ email: mail_addr, status: 'AlreadyExists' });
+                            log.info('Failed Adding user' + mail_addr + ' to department ' + req.params.department_id + ' ' + err);
+                        });
+                    }
+                }))).then(save_statuses => {
+                    res.status(200).json(result)
+                }).catch(err => { res.status(500).json({ message: err }) });
+            });
+        })
+        .catch(err => res.status(403).json(err));
+};
+
+///PUT volunteer/department/department_id/volunteers/user_id
+var put_volunteer = function (req, res) {
+    var new_data = {
+        role_id: req.body.role_id,
+        type_in_shift_id: req.body.shift_type,
+        comment: req.body.comment,
+        is_production: req.body.is_production
+    };
+    Volunteer
+        .where({ user_id: req.params.user_id, department_id: req.params.department_id, event_id: CURRENT_EVENT })
+        .save(new_data, { method: 'update', patch: true })
+        .then((vol) => {
+            res.status(200).json(vol.toJSON());
+        }).catch((err) => {
+            res.status(500).json({
+                error: true,
+                data: {
+                    message: err.message
+                }
+            })
+        });
+};
+
+var get_volunteering_info = function (req, res) {
+    var user_id = req.user.id;
```

Comment:
validate `req.user.id` here for existence. [_.get](https://lodash.com/docs/4.17.4#get)